plugin/src/modutils.ml

```diff
@@ -0,0 +1,97 @@
+open Util
+open Names
+open Globnames
+open Declarations
+open Indutils
+
+(*
+ * Pull any functor parameters off the module signature, returning the list of
+ * functor parameters and the list of module elements (i.e., fields).
+ *)
+let decompose_module_signature mod_sign =
+  let rec aux mod_arity mod_sign =
+    match mod_sign with
+    | MoreFunctor (mod_name, mod_type, mod_sign) ->
+      aux ((mod_name, mod_type) :: mod_arity) mod_sign
+    | NoFunctor mod_fields ->
+      mod_arity, mod_fields
+  in
+  aux [] mod_sign
+
+(*
+ * Define an interactive (i.e., elementwise) module structure, with the
+ * functional argument called to populate the module elements.
+ *
+ * The optional argument specifies functor parameters.
+ *)
+let declare_module_structure ?(params=[]) ident declare_elements =
```

Comment:
Same here, are a lot of these common with fix-to-elim? If so, we should move them to the library

Reply:
Same as the previous comment -- several of these helper functions are already found in the plugin library (`coq-plugin-lib`, specifically). Refactoring `modutils.ml` into it will deduplicate.